### PR TITLE
Improve the Semi-Abstracted nonce system

### PR DESCRIPTION
**Pull request by LiorAgnin**
Introduced a new property to the txOptions object, customNonceKey, which gives account holders greater control over the semi-abstracted nonce system.

### CODE REVIEW

File: example/batch.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const credentials = new ethers.Wallet(process.env.PRIVATE_KEY as string);
+  const publicApiKey = process.env.PUBLIC_API_KEY as string;
+  const fuseSDK = await FuseSDK.init(publicApiKey, credentials, { withPaymaster: true });
+
+  // You can use any other "to" address and any other "value"
+  const to = AddressZero;
+  const value = parseEther("0.001");
+  const data = Uint8Array.from([]);
+  const txOptions = { ...Variables.DEFAULT_TX_OPTIONS, useNonceSequence: true };
+  await Promise.all([
```
Comment (MusabShakeel576) [inline example/batch.ts]: @LiorAgnin we can just do "await", and remove `Promise.all`, as we are only calling "fuseSDK.executeBatch" once

Comment (LiorAgnin) [reply example/batch.ts]: Correct, thanks.

**Review by leonprou — APPROVED** [review]
Looks good
that way the clients can manage the nonceKey themselves 👍

**Review by MusabShakeel576 — APPROVED** [review]
LGTM

**Pull request merged**